### PR TITLE
Remove snapshot_loaded flag

### LINKED ISSUE #1089: Investigate Stronghold loaded flag
I wonder, if you really should go with a flag. One option would be assume Stronghold is being loaded and initialized. If access to certain secret paths fails, you can fall back to reload the Snapshot file from disk, or maybe I don't see the full picture here.

_Originally posted by @felsweg-iota in https://github.com/iotaledger/iota.rs/pull/1068#discussion_r901781917_

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Remove snapshot_loaded flag

## Links to any relevant issues

Fixes #1089 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Tests and running stronghold example from Python

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by thibault-martinez — COMMENTED** [review]
Can you add an entry in the changelog please ?

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 
 - Use NativeTokensBuilder::finish_vec to lift some unnecessary 64 Native Tokens limits;
+- Removed `snapshot_loaded` field from StrongholdAdapter;
```
Comment (thibault-martinez) [inline CHANGELOG.md:38]: Put that in a new `### Removed` section please

**Comment by thibault-martinez** [comment]
Conflicts

File: CHANGELOG.md (hunk 1 above)
Comment (samuel-rufi) [inline CHANGELOG.md:38]:
```suggestion

### Removed
- Removed `snapshot_loaded` field from StrongholdAdapter;
```

**Review by samuel-rufi — CHANGES_REQUESTED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**